src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -1044,13 +1059,13 @@ private int RunCore(TextWriter consoleOutput, ErrorLogger errorLogger, Cancellat
                             {
                                 // Apply diagnostic suppressions for analyzer and/or compiler diagnostics from diagnostic suppressors.
                                 analyzerDriver.ApplyProgrammaticSuppressions(diagnostics, compilation);
-
-                                if (HasUnsuppressedErrors(diagnostics))
-                                {
-                                    success = false;
-                                }
                             }
                         }
+
+                        if (HasReportedErrors(diagnostics))
```

Comment:
>HasReportedErrors [](start = 28, length = 17)

Could you remind where do we go through process of elevating/suppressing diagnostics? Can we be sure we went through that process at this point?

Reply:
For example, I see a couple of usages of ```HasUnsuppressedErrors``` API below. Should they be adjusted as well?

---
In reply to: [313566102](https://github.com/dotnet/roslyn/pull/37948#discussion_r313566102) [](ancestors = 313566102)

Reply:
Filtering of compiler diagnostics (i.e. applying the ruleset, nowarn, warnaserror, etc. options) happens at individual compilation stages (for example [here](https://github.com/dotnet/roslyn/blob/a5418f7398495db4ea858f0c9efd9b13541ff478/src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs#L2336)). After each stage, we bail out in presence of unsuppressable compiler errors(for example [here](https://github.com/dotnet/roslyn/blob/ad175163a22afe8a642fc8cad967dfa41d2458b8/src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs#L846) and [here](https://github.com/dotnet/roslyn/blob/ad175163a22afe8a642fc8cad967dfa41d2458b8/src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs#L894)), but not warnings escalated to errors as these could be programattically suppressed. Filtering of analyzer diagnostics happens immediately after an analyzer diagnostic is reported (for example [here](https://github.com/dotnet/roslyn/blob/master/src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs#L1477). At the end of compiling all method bodies, we force complete all analyzer execution and apply programatic suppressions [here](https://github.com/dotnet/roslyn/blob/master/src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs#L1037-L1052).

Reply:
> For example, I see a couple of usages of HasUnsuppressedErrors API below. Should they be adjusted as well?

I believe the checks in this method below are just optimizations to exit earlier from `CompileAndEmit`, so we can go either way. Caller immediately invokes [ReportDiagnostics](http://source.roslyn.io/#Microsoft.CodeAnalysis/CommandLine/CommonCompiler.cs,747) after invoking `CompileAndEmit`.